### PR TITLE
make sriov netdevice daemonset and configmap as a part of acc_provision 

### PR DESCRIPTION
**Pull request by Bhavanaashok33**

### CODE REVIEW

File: provision/testdata/.acirc [hunk 1]
```diff
@@ -1,2 +1,2 @@
-|public subnets|: [u'subnet-06cdb0b67bf86fcbf', u'subnet-0814cdfa1911664e7']
-|private subnets|: [u'subnet-093bf9479f50168f7', u'subnet-00a54d51ba268fe29']
+|public subnets|: ['subnet-06cdb0b67bf86fcbf', 'subnet-0814cdfa1911664e7']
```
Comment (snaiksat) [inline provision/testdata/.acirc:1]: nit: this file does not need to be part of the change-set

File: provision/acc_provision/acc_provision.py [hunk 2]
```diff
@@ -538,9 +541,33 @@ def config_adjust(args, config, prov_apic, no_random):
...
+        adj_config["devices"] = ""
+        device_info = get_device_info()
+        if device_info is not None:
+            adj_config["vendors"] = device_info[0]
+            adj_config["devices"] = device_info[1]
+
     return adj_config
 
 
+def get_device_info():
```
Comment (snaiksat) [inline provision/acc_provision/acc_provision.py:556]: I am not sure I understand this approach. This function is obtaining information from the host on which its running (you are probably making the assumption that it's a node in the cluster). However, acc-provision can be run from any place, and in fact, the cluster node won't even be ready when acc-provision is run. The job of acc-provision is to program the APIC and generate the ACI CNI deployment spec prior to the cluster install.

**Review by snaiksat — CHANGES_REQUESTED** [review]

**Comment by apoorva11029** [comment]
Hi @Bhavanaashok33 please squash your commits into one, since they are all correlated

File: provision/acc_provision/acc_provision.py (hunk 2 above)
Comment (Bhavanaashok33) [reply provision/acc_provision/acc_provision.py:556]: That makes sense. If we are not able to populate  vendor and device info of the Nic Cards as part of acc_provision run, we have to manually put in those information to  sriov network plugin configmap. Madhu and I will catch up with you to think about a better approach.

**Comment by Bhavanaashok33** [comment]
@apoorva11029 sure, I will squash it before I merge the changes.

File: provision/acc_provision/templates/overlay-provision-config.yaml [hunk 3]
```diff
@@ -64,3 +64,9 @@ istio_config:
 
 #drop_log_config:
   #enable: False        #default is True
+
+#sriov_config:
+  # enable: False       # default is False
```
Comment (snaiksat) [inline provision/acc_provision/templates/overlay-provision-config.yaml:69]: # enable: True

File: provision/acc_provision/templates/provision-config.yaml [hunk 4]
```diff
@@ -88,3 +88,9 @@ registry:
 
 #multus:
   # disable: False       # default is True
+
+#sriov_config:
+  # enable: False       # default is False
```
Comment (snaiksat) [inline provision/acc_provision/templates/provision-config.yaml:93]: # enable: True # default is False

File: provision/acc_provision/acc_provision.py [hunk 5]
```diff
@@ -538,6 +541,23 @@ def config_adjust(args, config, prov_apic, no_random):
         adj_config["aci_config"]["vmm_domain"]["injected_cluster_type"] = ""
     if not config["aci_config"]["vmm_domain"].get("injected_cluster_provider"):
         adj_config["aci_config"]["vmm_domain"]["injected_cluster_provider"] = ""
+
+    if config["sriov_config"].get("enable") is True:
```
Comment (snaiksat) [inline provision/acc_provision/acc_provision.py:545]: if config["sriov_config"].get("enable"):

File: provision/acc_provision/acc_provision.py [hunk 6]
```diff
@@ -538,6 +541,23 @@ def config_adjust(args, config, prov_apic, no_random):
...
+        if config["sriov_config"]["device_info"].get("devices") != None :
```
Comment (snaiksat) [inline provision/acc_provision/acc_provision.py:546]: if 'devices' in config["sriov_config"]["device_info"]:

File: provision/acc_provision/acc_provision.py [hunk 7]
```diff
@@ -538,6 +541,23 @@ def config_adjust(args, config, prov_apic, no_random):
...
+            adj_config["devices"] = str(config["sriov_config"]["device_info"].get("devices"))
+        else:
+            adj_config["devices"] = ""
+
+        if config["sriov_config"]["device_info"].get("isRdma") is True:
```
Comment (snaiksat) [inline provision/acc_provision/acc_provision.py:551]: if config["sriov_config"]["device_info"].get("isRdma):

**Review by snaiksat — CHANGES_REQUESTED** [review]

File: provision/acc_provision/templates/overlay-provision-config.yaml (hunk 3 above)
Comment (Bhavanaashok33) [reply provision/acc_provision/templates/overlay-provision-config.yaml]: resolved

File: provision/acc_provision/templates/provision-config.yaml (hunk 4 above)
Comment (Bhavanaashok33) [reply provision/acc_provision/templates/provision-config.yaml]: addressed

File: provision/acc_provision/acc_provision.py (hunk 5 above)
Comment (Bhavanaashok33) [reply provision/acc_provision/acc_provision.py]: addressed

File: provision/acc_provision/acc_provision.py (hunk 6 above)
Comment (Bhavanaashok33) [reply provision/acc_provision/acc_provision.py]: resolved

File: provision/acc_provision/acc_provision.py (hunk 7 above)
Comment (Bhavanaashok33) [reply provision/acc_provision/acc_provision.py]: resolved

**Comment by Bhavanaashok33** [comment]
I am going to close this PR and open a new PR with the same changes. I had issues while squashing the multiple commits into one commit due merge pull requests from master branch to the local branch.. Please leave your comments on the new PR if any.. Thanks

**Pull request closed without merging**